src/openpai-runtime/build/openpai-runtime.k8s.dockerfile

```diff
@@ -15,5 +15,5 @@
 # DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-FROM openpairuntime/openpai-runtime:v0.1.1
+FROM openpairuntime/openpai-runtime:latest
```

Comment:
>openpairuntime/openpai-runtime:latest [](start = 5, length = 37)

Why use latest instead of a fixed version? If so, how we define a PAI version, it cannot be fixed as the runtime is not fixed here

Reply:
It's a temporary change. I will use fixed before PAI next release.  In my plan, runtime may bring more features in the next release, will change this line after new runtime release is published

Reply:
But even in master branch or by yourself debugging/testing, it is also hard for you to figure current version when sth bad happens.

Reply:
In my point of view. The `latest` means the latest code version. I will update the runtime docker image after each commit. So if something bad happens, the bug is in the latest code. (I am looking for a way to build docker image automatically through docker hub after each commit).

For `debug/testing` purpose, I could use another tag, but after runtime code commit, it's better to use `latest` tag in PAI repo.

Maybe the process can be PAI-master->runtime:latest in devolving phrase.
                                           PAI-release-branch->runtime:release in Testing phrase

Reply:
It seems to me we need to update the build process. Why do we build the openpai-runtime in this repo, but not in https://github.com/microsoft/openpai-runtime?

Reply:
The build pipeline can fail.
Anyway, we should make sure the **master** branch code is also reproducible at any time given a fixed PAI git version.

Reply:
@fanyangCS 
Build the openpai-runtime in this repo will give customer the ability to customize the registry mirror, and avoid pulling image failure.

@yqwang-ms 
Agree should make **master** branch be reproducible. Maybe we can use git `submodules` to solve this issue.

Another reason for this issue is runtime and rest-server are tightly coupled, there is no clear interface, and change one may break the other. Maybe give clear interface will solve this issue.

And for this case, current implement also tightly coupled, if we can make runtime be more flexible. Then we can implement this feature in PAI repo actually.
Such as provide rest-server change and a runtime plugin in PAI side.
When pai build runtime image, refer the `openpairuntime/openpai-runtime` image, copy plugin to certain folder. Then pai will have a customized runtime.

For `openpai-runtime` repo just keep common plugins and main logic.

Reply:
Wait for runtime release, here will change